addon/ng2/blueprints/component/index.js

```diff
@@ -16,6 +27,11 @@ module.exports = {
     var parsedPath = dynamicPathParser(this.project, entityName);
 
     this.dynamicPath = parsedPath;
+    
+    if (!validateName(parsedPath.name)) {
+      throw 'Invalid name. Names must be camelCase or kebab-case';
```

Comment:
The purpose of this is to enforce the selector to contain a `-` so I agree, I'll tweak it to show a more useful message to the user.

I'm aware that the names can contain any variety of capital/lower/dash characters, but that is indeed intentional, as long as the name transformation creates a selector with at least 1 dash, then it will satisfy the HTML spec for custom elements.